### PR TITLE
Added an LFSR proc example

### PR DESCRIPTION
**Pull request by mdagois**
Added a new example that wraps an 8-bit LFSR into a proc.
The user can set the LFSR seed or get the next value in the sequence.

### CODE REVIEW

File: xls/examples/lfsr_proc.x [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  }
+
+  config(output_s: chan<u8> out, seed_r: chan<u8> in) {
+    (output_s, seed_r)
+  }
+
+  next(tok: token, state: u8) {
+    let (tok, seed, seed_valid) = recv_non_blocking(tok, seed_r, state);
+    let state = if(seed_valid) {
+        (seed)
```
Comment (proppy) [inline xls/examples/lfsr_proc.x:39]: actually I'm not sure if you need this, as the `recv_non_blocking` will default to `state` if there is nothing to `recv`?
you could simply `send(seed)` and return `lfsr::lfsr` as the new state.

wdyt?

File: xls/examples/lfsr_proc.x [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+// 8-bit LFSR proc
+// 
+// Two operations are supported:
+// 1. setting the seed
+// 2. getting the next value from the LFSR counter
+////////////////////////////////////////////////////////////////////////////////
+
+import xls.examples.lfsr
+
+proc user_module {
```
[the remark this answers is not in the document]
Comment (mdagois) [reply xls/examples/lfsr_proc.x]: I turned to proc into a parametric proc. It takes the bit width of the LFSR as parameter now.

File: xls/examples/lfsr_proc.x [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  config(output_s: chan<u8> out, seed_r: chan<u8> in) {
+    (output_s, seed_r)
+  }
+
+  next(tok: token, state: u8) {
+    let (tok, seed, seed_valid) = recv_non_blocking(tok, seed_r, state);
+    let state = if(seed_valid) {
+        (seed)
+    } else {
+        lfsr::lfsr(state, u8:0b10111000)
```
[the remark this answers is not in the document]
Comment (mdagois) [reply xls/examples/lfsr_proc.x]: After turning the proc into a parametric one, it basically became mandatory to do so. :)

File: xls/examples/lfsr_proc.x (hunk 1 above)
Comment (mdagois) [reply xls/examples/lfsr_proc.x]: Ah yes, you are right. I simplified the code.

**Pull request merged**